File: string/length-of-last-word/length-of-last-word.ts

function lengthOfLastWord(s: string): number {
    let tail = s.length - 1;
    while (tail >= 0 && s[tail] === ' ') tail--;
    let ans = 0;
    while (tail >= 0 && s[tail] !== ' ') {
        ans++;
        tail--;
    };

    return ans;
}

function easyLengthOfLastWord(s: string): number {
    const w = s.trim().split(/\s/)
    return w[w.length - 1].length;
}

File: string/length-of-last-word/length-of-last-word.go
package lengthoflastword

import "strings"

func lengthOfLastWord(s string) int {
	tail := len(s) - 1
	for tail >= 0 && s[tail] == ' ' {
		tail--
	}
	ans := 0
	for tail >= 0 && s[tail] != ' ' {
		ans++
		tail--
	}

	return ans
}

func easyLengthOfLastWord(s string) int {
	w := strings.Fields(s)
	return len(w[len(w)-1])
}
